packages/material-ui/src/Select/SelectInput.test.js

```diff
@@ -183,292 +247,429 @@ describe('<SelectInput />', () => {
       clock.restore();
     });
 
-    beforeEach(() => {
-      handleChange = spy();
-      wrapper = mount(
+    it('should call onChange when clicking an item', () => {
+      const handleChange = stub().callsFake(event => event.target.value);
+      const { getByRole, getAllByRole, queryByRole } = render(
         <SelectInput
-          {...defaultProps}
-          onChange={handleChange}
+          classes={{}}
+          IconComponent="div"
           MenuProps={{ transitionDuration: 0 }}
-        />,
+          onChange={handleChange}
+          value=""
+        >
+          <MenuItem value="">none</MenuItem>
+          <MenuItem value={10}>Ten</MenuItem>
+        </SelectInput>,
       );
-    });
 
-    it('should call onChange when clicking an item', () => {
-      wrapper.find(`.${defaultProps.classes.select}`).simulate('click');
-      assert.strictEqual(wrapper.find(MenuItem).exists(), true);
-      wrapper
-        .find('li')
-        .at(1)
-        .simulate('click');
-      clock.tick(0);
-      wrapper.update();
-      assert.strictEqual(wrapper.find(MenuItem).exists(), false);
-      assert.strictEqual(handleChange.callCount, 1);
-      assert.strictEqual(handleChange.args[0][0].target.value, 20);
+      fireEvent.click(getByRole('button'));
+      fireEvent.click(getAllByRole('option')[1]);
+      act(() => {
+        clock.tick(0);
+      });
+
+      expect(handleChange.calledOnce).to.be.true;
+      expect(handleChange.firstCall.returnValue).to.equal(10);
+      expect(queryByRole('listbox')).to.be.null;
     });
 
     it('should ignore onBlur the first time the menu is open', () => {
```

Comment:
This is IMO a leaky abstraction. The point of this is that we don't blindly close the select if  the trigger is blurred. You shouldn't manually listen to DOM events to control open anyway. That's what `onClose` is for.

Since this isn't documented and we don't have an official stance on public vs internal tests I would remove this specific test while ensuring a userclick on the button does not close because the trigger was blurred which will be covered by https://github.com/mui-org/material-ui/pull/17299